nucypher/datastore/ritual.py

```diff
@@ -1,23 +1,81 @@
+import abc
+from abc import abstractmethod
 from collections import defaultdict
 from typing import List, Optional
 
 from atxm.tx import AsyncTx
-from nucypher_core.ferveo import (
-    Validator,
-)
+from nucypher_core.ferveo import Validator
 
 from nucypher.blockchain.eth.models import (
+    DKG_PHASE_1,
+    DKG_PHASE_2,
     HANDOVER_AWAITING_BLINDED_SHARE,
     HANDOVER_AWAITING_TRANSCRIPT,
-    PHASE1,
-    PHASE2,
+    SIGNING_AWAITING_SIGNATURES,
     Coordinator,
 )
-from nucypher.types import PhaseId
+from nucypher.types import PhaseId, PhaseNumber
+
+
+class RitualStorage(abc.ABC):
+    """A simple in-memory storage for ritual data"""
+
+    _KEYS: List[str] = NotImplemented
+
+    def __init__(self):
+        self._data = defaultdict(dict)
+
+    def clear(self, ritual_id):
+        for key in self._KEYS:
+            try:
+                del self._data[key][ritual_id]
+            except KeyError:
+                continue
+
+    #
+    # DKG Phases
+    #
+    @classmethod
+    @abstractmethod
+    def _get_phase_key(cls, phase: PhaseNumber):
+        raise NotImplementedError
+
+    def store_ritual_phase_async_tx(self, phase_id: PhaseId, async_tx: AsyncTx):
+        key = self._get_phase_key(phase_id.phase)
+        self._data[key][phase_id.ritual_id] = async_tx
 
+    def clear_ritual_phase_async_tx(self, phase_id: PhaseId, async_tx: AsyncTx) -> bool:
+        key = self._get_phase_key(phase_id.phase)
+        if self._data[key].get(phase_id.ritual_id) is async_tx:
+            try:
+                del self._data[key][phase_id.ritual_id]
+                return True
+            except KeyError:
+                pass
+        return False
+
+    def get_ritual_phase_async_tx(self, phase_id: PhaseId) -> Optional[AsyncTx]:
+        key = self._get_phase_key(phase_id.phase)
+        return self._data[key].get(phase_id.ritual_id)
```

Comment:
Same suggestion than the one above

```suggestion
    def get_ritual_phase_async_tx(self, phase_id: PhaseId) -> Optional[AsyncTx]:
        key = self._get_phase_key(phase_id.phase)
        return self._get_ritual_metadata(key, phase_id.ritual_id)
```

Reply:
✔️